src/SFA.DAS.ApprenticeCommitments.Api/Controllers/GetRequest.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeCommitments.Api.Controllers
+{
+    public class GetRequest<TResponse> : IGetApiRequest
+    {
+        public string GetUrl { get; }
+
+        public GetRequest(string url) => GetUrl = url;
+
+        public async Task<IActionResult> Get<TClient>(IInternalApiClient<TClient> client)
+        {
+            var response = await client.GetWithResponseCode<TResponse>(this);
+
+            if (!string.IsNullOrEmpty(response.ErrorContent))
+            {
+                return new ObjectResult(response.ErrorContent)
+                {
+                    StatusCode = (int)response.StatusCode
+                };
+            }
+
+            return new ObjectResult(response.Body)
+            {
+                StatusCode = (int)response.StatusCode
+            };
+        }
+    }
+}
```

Comment:
Does this belong to the shared IInternalApiClient code base. As it may be needed by all